internal/provider/kubernetes/rbac.go

```diff
@@ -3,4 +3,7 @@ package kubernetes
 // +kubebuilder:rbac:groups="gateway.networking.k8s.io",resources=gatewayclasses;gateways;httproutes;referencepolicies;referencegrants,verbs=get;list;watch
 // +kubebuilder:rbac:groups="gateway.networking.k8s.io",resources=gatewayclasses/status;gateways/status;httproutes/status,verbs=update
 
-// +kubebuilder:rbac:groups="",resources=secrets;services;namespaces,verbs=get;list;watch
+// RBAC for Infra Manager to manage Envoy.
+// +kubebuilder:rbac:groups="",resources=secrets;namespaces,verbs=get;list;watch
+// +kubebuilder:rbac:groups="",resources=services;serviceaccounts,verbs=get;list;watch;create;update;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;delete
```

Comment:
Do we want to restrict some of these just to the default `envoy-gateway-system` namespace since that's where we're creating all infra? Feels better than giving CRUD access across all namespaces. It does mean that if a user wants to deploy EG into an alternate namespace, they'll need to customize the generated RBAC, but to me that's a better trade-off than not restricting them.

Reply:
@skriss https://github.com/envoyproxy/gateway/pull/292 restricts Infra Manager RBAC to a Role within the envoy-gateway-system namespace instead of a ClusterRole. The Infra Manager RBAC is static but that should do for v0.2.0 unless someone wants to pick up https://github.com/envoyproxy/gateway/issues/289.